### PR TITLE
Improve docstrings for controlled common gate overrides

### PR DESCRIPTION
**Pull request by dabacon**
Follow up to #3113 to improve the docstrings.

### CODE REVIEW

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -122,23 +122,32 @@ def controlled(self,
...
-
-        This behavior only occurs when the last control qubit is a default-type
-        control qubit. A default-type control qubit is one with shape of 2 (not
-        a generic qudit) and where the control is satisfied by the qubit being
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CXPowGate (or controlled version of that gate) if the global shift on
-        the XPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CXPowGate is, by definition, a controlled-XPowGate.)
+        Returns a `CXPowGate` or a controlled `CXPowGate` when possible.
```
Comment (balopat) [inline cirq/ops/common_gates.py:125]:
```suggestion
        Returns a `CXPowGate` or a controlled `XPowGate` when possible.
```

File: cirq/ops/common_gates.py [hunk 2]
```diff
@@ -443,23 +452,32 @@ def controlled(self,
...
-
-        This behavior only occurs when the last control qubit is a default-type
-        control qubit. A default-type control qubit is one with shape of 2 (not
-        a generic qudit) and where the control is satisfied by the qubit being
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CZPowGate (or controlled version of that gate) if the global shift of
-        the ZPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CZPowGate is, by definition, a controlled-ZPowGate.)
+        Returns a `CZPowGate` or a controlled `CZPowGate` when possible.
```
Comment (balopat) [inline cirq/ops/common_gates.py:455]:
```suggestion
        Returns a `CZPowGate` or a controlled `ZPowGate` when possible.
```

File: cirq/ops/common_gates.py [hunk 3]
```diff
@@ -966,23 +993,32 @@ def controlled(self,
...
-
-        This behavior only occurs when the last control qubit is a default-type
-        control qubit. A default-type control qubit is one with shape of 2 (not
-        a generic qudit) and where the control is satisfied by the qubit being
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CCXPowGate (or controlled version of that gate) if the global shift of
-        the CXPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CCXPowGate is, by definition, a controlled-CXPowGate.)
+        Returns a `CCXPowGate` or a controlled `CCXPowGate` when possible.
```
Comment (balopat) [inline cirq/ops/common_gates.py:996]:
```suggestion
        Returns a `CCXPowGate` or a controlled `CXPowGate` when possible.
```

File: cirq/ops/common_gates.py [hunk 4]
```diff
@@ -966,23 +993,32 @@ def controlled(self,
...
+
+        The `controlled` method of the `Gate` class, of which this class is a
+        child, returns a `ControlledGate`. This method overrides this behavior
+        to return a `CCXPowGate` or a `ControlledGate` of a `CCXPowGate, when
```
Comment (balopat) [inline cirq/ops/common_gates.py:1000]:
```suggestion
        to return a `CCXPowGate` or a `ControlledGate` of a `CXPowGate`, when
```

File: cirq/ops/common_gates.py [hunk 5]
```diff
@@ -122,23 +122,32 @@ def controlled(self,
...
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CXPowGate (or controlled version of that gate) if the global shift on
-        the XPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CXPowGate is, by definition, a controlled-XPowGate.)
+        Returns a `CXPowGate` or a controlled `CXPowGate` when possible.
+
+        The `controlled` method of the `Gate` class, of which this class is a
+        child, returns a `ControlledGate`. This method overrides this behavior
+        to return a `CXPowGate` or a `ControlledGate` of a `CXPowGate, when
```
Comment (balopat) [inline cirq/ops/common_gates.py:129]:
```suggestion
        to return a `CXPowGate` or a `ControlledGate` of a `XPowGate, when
```

File: cirq/ops/common_gates.py [hunk 6]
```diff
@@ -122,23 +122,32 @@ def controlled(self,
...
+            * The `control_values` and `control_qid_shape` are compatible with
+                the `CXPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CXPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CXPowGate`. In the
```
Comment (balopat) [inline cirq/ops/common_gates.py:143]:
```suggestion
        `ControlledGate` with the `Gate` being a `XPowGate`. In the
```

File: cirq/ops/common_gates.py [hunk 7]
```diff
@@ -443,23 +452,32 @@ def controlled(self,
...
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CZPowGate (or controlled version of that gate) if the global shift of
-        the ZPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CZPowGate is, by definition, a controlled-ZPowGate.)
+        Returns a `CZPowGate` or a controlled `CZPowGate` when possible.
+
+        The `controlled` method of the `Gate` class, of which this class is a
+        child, returns a `ControlledGate`. This method overrides this behavior
+        to return a `CZPowGate` or a `ControlledGate` of a `CZPowGate`, when
```
Comment (balopat) [inline cirq/ops/common_gates.py:459]:
```suggestion
        to return a `CZPowGate` or a `ControlledGate` of a `ZPowGate`, when
```

File: cirq/ops/common_gates.py [hunk 8]
```diff
@@ -966,23 +993,32 @@ def controlled(self,
...
+            * The `control_values` and `control_qid_shape` are compatible with
+                the `CCXPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CCXPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CCXPowGate`. In the
```
Comment (balopat) [inline cirq/ops/common_gates.py:1014]:
```suggestion
        `ControlledGate` with the `Gate` being a `CXPowGate`. In the
```

File: cirq/ops/common_gates.py [hunk 9]
```diff
@@ -966,23 +993,32 @@ def controlled(self,
...
+        later case the `ControlledGate` is controlled by one less qudit
```
Comment (balopat) [inline cirq/ops/common_gates.py:1015]:
```suggestion
        latter case the `ControlledGate` is controlled by one less qudit
```

File: cirq/ops/common_gates.py [hunk 10]
```diff
@@ -122,23 +122,32 @@ def controlled(self,
...
+                the `CXPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CXPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CXPowGate`. In the
+        later case the `ControlledGate` is controlled by one less qudit
```
Comment (balopat) [inline cirq/ops/common_gates.py:144]:
```suggestion
        latter case the `ControlledGate` is controlled by one less qudit
```

File: cirq/ops/common_gates.py [hunk 11]
```diff
@@ -443,23 +452,32 @@ def controlled(self,
...
+                the `CZPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CZPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CZPowGate`. In the
+        later case the `ControlledGate` is controlled by one less qudit
```
Comment (balopat) [inline cirq/ops/common_gates.py:474]:
```suggestion
        latter case the `ControlledGate` is controlled by one less qudit
```

File: cirq/ops/common_gates.py [hunk 12]
```diff
@@ -797,23 +815,32 @@ def controlled(self,
...
+                the `CCZPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CCZPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CCZPowGate`. In the
+        later case the `ControlledGate` is controlled by one less qudit
```
Comment (balopat) [inline cirq/ops/common_gates.py:837]:
```suggestion
        latter case the `ControlledGate` is controlled by one less qudit
```

File: cirq/ops/common_gates.py [hunk 13]
```diff
@@ -443,23 +452,32 @@ def controlled(self,
...
+        If these conditions are met, then the returned object is a `CZPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CZPowGate`. In the
+        later case the `ControlledGate` is controlled by one less qudit
+        than specified in `control_values` and `control_qid_shape` (since
+        one of these, the last qubit, is used as the control for the
+        `CZPowGate`).
+
+        If the above conditions are not met, a `ControlledGate` of this
+        gate will be returned..
```
Comment (balopat) [inline cirq/ops/common_gates.py:480]:
```suggestion
        gate will be returned.
```

File: cirq/ops/common_gates.py [hunk 14]
```diff
@@ -797,23 +815,32 @@ def controlled(self,
...
-
-        This behavior only occurs when the last control qubit is a default-type
-        control qubit. A default-type control qubit is one with shape of 2 (not
-        a generic qudit) and where the control is satisfied by the qubit being
-        ON, as opposed to OFF. Note also that this only transforms into a
-        CCZPowGate (or controlled version of that gate) if the global shift of
-        the CZPowGate is 0, otherwise it produces a normal ControlledGate.
-
-        (Note that a CCZPowGate is, by definition, a controlled-CZPowGate.)
+        Returns a `CCZPowGate` or a controlled `CCZPowGate` when possible.
```
Comment (balopat) [inline cirq/ops/common_gates.py:818]:
```suggestion
        Returns a `CCZPowGate` or a controlled `CZPowGate` when possible.
```

File: cirq/ops/common_gates.py [hunk 15]
```diff
@@ -797,23 +815,32 @@ def controlled(self,
...
+
+        The `controlled` method of the `Gate` class, of which this class is a
+        child, returns a `ControlledGate`. This method overrides this behavior
+        to return a `CCZPowGate` or a `ControlledGate` of a `CCZPowGate, when
```
Comment (balopat) [inline cirq/ops/common_gates.py:822]:
```suggestion
        to return a `CCZPowGate` or a `ControlledGate` of a `CZPowGate`, when
```

File: cirq/ops/common_gates.py [hunk 16]
```diff
@@ -797,23 +815,32 @@ def controlled(self,
...
+            * The `control_values` and `control_qid_shape` are compatible with
+                the `CCZPowGate`:
+                * The last value of `control_qid_shape` is a qubit.
+                * The last value of `control_values` corresponds to the
+                    control being satisfied if that last qubit is 1 and
+                    not satisfied if the last qubit is 0.
+
+        If these conditions are met, then the returned object is a `CCZPowGate`
+        or, in the case that there is more than one controlled qudit, a
+        `ControlledGate` with the `Gate` being a `CCZPowGate`. In the
```
Comment (balopat) [inline cirq/ops/common_gates.py:836]:
```suggestion
        `ControlledGate` with the `Gate` being a `CZPowGate`. In the
```

File: cirq/ops/common_gates.py (hunk 1 above)
Comment (smitsanghavi) [reply cirq/ops/common_gates.py:125]: I think this (and others below) might be intentionally referring to cases where there are multiple controls which would result in a controlled `CXPowGate`. But I might be wrong. Would also like to know if there are other suggestions on the wording here.

File: cirq/ops/common_gates.py (hunk 5 above)
Comment (smitsanghavi) [reply cirq/ops/common_gates.py:129]: Same as above. Also need to add ` before "," here and below.

File: cirq/ops/common_gates.py (hunk 1 above)
Comment (balopat) [reply cirq/ops/common_gates.py:125]: I think you're right, thanks. I'll remove my noisy comments :) 
I'm going to look at the actual code and try to understand what's happening.

Comment (balopat) [reply cirq/ops/common_gates.py:125]: If I understand correctly the whole idea here is that we have named CX and CCX gates, let's convert them when we can. 

What do you think about this? A bit wordy.

```
Returns a Controlled XPowGate and, if possible, uses the named `CXPowGate` at the end of the control chain.
```

Comment (dabacon) [reply cirq/ops/common_gates.py:125]: How about
```
Returns a controlled `XPowGate`, using a `CXPowGate` where possible.
```
I think this is one of those cases where the top most string cannot carry the weight of the exceptions, but we want to warn the reader that they need to read more.  This accurately describes the result of the method, and say "but watchout!"  The watchout stuff is then described in the rest of the docstring.

Comment (balopat) [reply cirq/ops/common_gates.py:125]: This version sounds good to me!

**Comment by dabacon** [comment]
PTAL.  Sorry I missed that you had suggested edits, so did those in my own branch :)

**Comment by balopat** [comment]
> PTAL. Sorry I missed that you had suggested edits, so did those in my own branch :)

No worries! I just like to use the suggested edits for typos because it shows the diff better. I don't usually like it to be committed - it creates too many commits, mixes authors, etc.

**Comment by dabacon** [comment]
@googlebot rescan please
